### PR TITLE
Remove isar steps in flotilla

### PR DESCRIPTION
**Pull request by mrica-equinor**

### CODE REVIEW

File: .github/workflows/backend_lint_and_test.yml [hunk 1]
```diff
@@ -59,8 +59,3 @@ jobs:
       # We are using severity level 'info' here.
       - name: Run dotnet format
         run: dotnet format --severity info --verbosity diagnostic --verify-no-changes --exclude ./api/migrations
-
-      - name: Run csharpier format
-        run: |
-          dotnet tool restore
-          dotnet csharpier --check
```
Comment (oysand) [inline .github/workflows/backend_lint_and_test.yml:66]: Why is this removed?

Comment (mrica-equinor) [reply .github/workflows/backend_lint_and_test.yml:66]: It has been failing in the last PRs so we decided to remove it and create an issue to be reintroduced later

File: backend/api/Services/MissionDefinitionService.cs [hunk 2]
```diff
@@ -185,8 +185,8 @@ private IQueryable<MissionDefinition> GetMissionDefinitionsWithSubModels(bool re
                 .Include(missionDefinition => missionDefinition.Source)
                 .Include(missionDefinition => missionDefinition.LastSuccessfulRun)
                 .ThenInclude(missionRun => missionRun != null ? missionRun.Tasks : null)!
-                .ThenInclude(missionTask => missionTask.Inspections)
-                .ThenInclude(inspection => inspection.InspectionFindings)
+                .ThenInclude(missionTask => missionTask.Inspection)
+                .ThenInclude(inspection => inspection != null ? inspection.InspectionFindings : null)!
```
Comment (oysand) [inline backend/api/Services/MissionDefinitionService.cs:189]: What do the ! at the end do? Is it necessary?

File: backend/api/Database/Models/MissionTask.cs [hunk 3]
```diff
@@ -40,13 +40,17 @@ public MissionTask(
...
             RobotPose = taskQuery.RobotPose;
             TaskOrder = taskQuery.TaskOrder;
             Status = TaskStatus.NotStarted;
-            Type = MissionTaskType.Inspection;
+            Type = MissionTaskType.ReturnHome;
+            if (taskQuery.Inspection is not null)
+            {
+                Inspection = new Inspection((CustomInspectionQuery)taskQuery.Inspection);
+                Type = MissionTaskType.Inspection;
+            }
```
Comment (tsundvoll) [inline backend/api/Database/Models/MissionTask.cs:53]: Maybe rewrite to an else clause, to be more explicit:

```cs
if (taskQuery.Inspection is not null)
{
    Inspection = new Inspection((CustomInspectionQuery)taskQuery.Inspection);
    Type = MissionTaskType.Inspection;
}
else
{
    Type = MissionTaskType.ReturnHome;
}
```

File: .github/workflows/backend_lint_and_test.yml (hunk 1 above)
Comment (andchiind) [reply .github/workflows/backend_lint_and_test.yml:66]: We can now rebase from main and re-add it since it has been fixed

File: backend/api/Database/Models/Inspection.cs [hunk 4]
```diff
@@ -44,7 +44,7 @@ public Inspection(CustomInspectionQuery inspectionQuery)
         public Inspection(Inspection copy, InspectionStatus? inspectionStatus = null, bool useEmptyIDs = false)
```
Comment (andchiind) [inline backend/api/Database/Models/Inspection.cs:44]:
```suggestion
        public Inspection(Inspection copy, InspectionStatus? inspectionStatus = null, bool useEmptyID = false)
```
Very pedantic sorry 😬

Comment (andchiind) [reply backend/api/Database/Models/Inspection.cs:44]: Also, have you tested that this (not using an empty IsarTaskId) does not affect the MissionTask hash generation code?

File: backend/api/Database/Models/MissionRun.cs [hunk 5]
```diff
@@ -161,7 +161,7 @@ public void CalculateEstimatedDuration()
                 );
                 foreach (var task in Tasks)
                 {
-                    numberOfTags += task.Inspections.Count;
+                    numberOfTags += 1;
```
Comment (andchiind) [inline backend/api/Database/Models/MissionRun.cs:164]: This variable can be deleted

File: backend/api/Database/Models/MissionTask.cs [hunk 6]
```diff
@@ -28,9 +28,7 @@ public MissionTask(
             TaskStatus status = TaskStatus.NotStarted,
             MissionTaskType type = MissionTaskType.Inspection)
         {
-            Inspections = inspections
-                .Select(inspection => new Inspection(inspection))
-                .ToList();
+            Inspection = inspection;
```
Comment (andchiind) [inline backend/api/Database/Models/MissionTask.cs:31]:
```suggestion
            Inspection = new Inspection(inspection);
```

File: backend/api/Database/Models/MissionTask.cs [hunk 7]
```diff
@@ -152,17 +153,14 @@ or TaskStatus.Failed
...
         public DateTime? EndTime { get; private set; }
 
-        public IList<Inspection> Inspections { get; set; }
+        public Inspection? Inspection { get; set; }
 
         public void UpdateWithIsarInfo(IsarTask isarTask)
         {
             UpdateStatus(isarTask.TaskStatus);
-            foreach (var inspection in Inspections)
+            if (isarTask.TaskType != IsarTaskType.ReturnToHome && isarTask.TaskType != IsarTaskType.Localize && isarTask.TaskType != IsarTaskType.MoveArm)
```
Comment (andchiind) [inline backend/api/Database/Models/MissionTask.cs:161]:
```suggestion
            if (![IsarTaskType.ReturnToHome, IsarTaskType.Localize, IsarTaskType.MoveArm].contains(isarTask.TaskType))
```
Subjective change, feel free to ignore

File: backend/api/Database/Models/MissionTask.cs [hunk 8]
```diff
@@ -210,9 +199,15 @@ public static string CalculateHashFromTasks(IList<MissionTask> tasks)
...
                     Id = "",
-                    IsarTaskId = ""
+                    IsarTaskId = "",
                 };
-                taskCopy.Inspections = taskCopy.Inspections.Select(i => new Inspection(i, useEmptyIDs: true)).ToList();
+                if (taskCopy.Inspection is not null)
+                {
+                    taskCopy.Inspection = new Inspection(taskCopy.Inspection, useEmptyIDs: true)
+                    {
+                        IsarTaskId = ""
```
Comment (andchiind) [inline backend/api/Database/Models/MissionTask.cs:208]: Why did we remove this functionality from the use of the useEmptyIDs flag in the constructor if we readd that functionality here?

File: backend/api/Services/EchoService.cs [hunk 9]
```diff
@@ -262,25 +262,36 @@ List<EchoPlantInfoResponse> echoPlantInfoResponse
...
-            (
-                inspections: echoTag.Inspections
+            var inspections = echoTag.Inspections
                     .Select(inspection => new Inspection(
                         inspectionType: inspection.InspectionType,
                         videoDuration: inspection.TimeInSeconds,
                         inspection.InspectionPoint,
                         status: InspectionStatus.NotStarted))
-                    .ToList(),
+                    .ToList();
```
Comment (andchiind) [inline backend/api/Services/EchoService.cs:273]: Can we remove one indentation from this line and the ones before it as well?

File: backend/api/Services/EchoService.cs [hunk 10]
```diff
@@ -262,25 +262,36 @@ List<EchoPlantInfoResponse> echoPlantInfoResponse
...
+
+            var missionTasks = new List<MissionTask>();
+
+            foreach (var inspection in inspections)
+            {
+                missionTasks.Add(
+                    new MissionTask
+            (
+                inspection: inspection,
```
Comment (andchiind) [inline backend/api/Services/EchoService.cs:282]: It might be just the github presentation, but this indentation also looks weird

File: frontend/src/models/Inspection.ts [hunk 11]
```diff
@@ -9,7 +9,7 @@ export interface Inspection {
     inspectionTarget: Position
     videoDuration?: number
     analysisType?: string
-    inspectionUrl?: string
+    inspectionUrl?: string | null
```
Comment (andchiind) [inline frontend/src/models/Inspection.ts:12]: I'm probably misunderstanding something about typescript, but isn't this implied with the question mark? Or does the question mark indicate "| undefined"?

**Review by andchiind — COMMENTED** [review]
Looks very good otherwise

File: frontend/src/models/Inspection.ts (hunk 11 above)
Comment (mrica-equinor) [reply frontend/src/models/Inspection.ts:12]: This was me trying to debug something, good catch thanks!

File: backend/api/Database/Models/MissionTask.cs (hunk 7 above)
Comment (mrica-equinor) [reply backend/api/Database/Models/MissionTask.cs:161]: This didn't work - There is no target type for the collection expression.[CS9176]

**Review by andchiind — APPROVED** [review]
LGTM
